### PR TITLE
[WIP] Add new class PromMetrics to stream old prometheus data to an observable

### PR DESCRIPTION
**Pull request by 4n4nd**
Rename old class from PromMetrics to PromMetricsLive
Update Prometheus class
Update app.py to use the new PromMetrics class
Update Pipfile and Pipfile lock

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -4,12 +4,17 @@
 
 
 if __name__ == '__main__':
-    metrics = metrics.FileMetrics() # this is an observable that streams in all the data alerts->etcd->build
+    metrics_list = ['openshift_build_info', 'etcd_object_counts', 'alerts']
```
Comment (durandom) [inline app.py]: let's read the list of metrics from an ENV variable. And if that variable exists, then setup `metrics` in the next line

**Review by durandom — CHANGES_REQUESTED** [review]

File: app.py (hunk 1 above)
Comment (4n4nd) [reply app.py]: Changed it in the new commit. Should I read other variables from env as well (vars like start/end_datetime and chunk_size)?

File: app.py [hunk 2]
```diff
@@ -1,15 +1,30 @@
 import flatliners
 import metrics
 import os
+import sys
 
 
 if __name__ == '__main__':
-    metrics = metrics.FileMetrics() # this is an observable that streams in all the data alerts->etcd->build
+    metrics_list = os.getenv("METRICS_LIST")
```
Comment (durandom) [inline app.py]: prefix with `FLT_` like [here](https://github.com/AICoE/prometheus-flatliner/blob/master/app.py#L54)

File: app.py [hunk 3]
```diff
@@ -1,15 +1,30 @@
...
+    if metrics_list:
+        metrics_list = str(metrics_list).split(",")
+        print("The metrics initialized were: ",metrics_list)
+        metric_start_datetime = os.getenv("METRIC_START_DATETIME","14 Sept 2018")
```
Comment (durandom) [inline app.py]: same here, prefix all env vars with `FLT_`

File: metrics/prommetrics.py [hunk 4]
```diff
@@ -23,43 +29,35 @@ def connect(self):
...
+        self.observe_prom_metrics_range(observer=observer,
+                                        metrics_list=self.metrics_list,
+                                        start_time=self.metric_start_datetime,
+                                        end_time=self.metric_end_datetime,
+                                        chunk_size=self.metric_chunk_size)
 
-    def _get_metrics_from_prometheus(self, observer=None):
+    def observe_prom_metrics_range(self, observer, metrics_list, start_time, end_time='now', chunk_size='1h'):
         # Collect credentials to connect to a prometheus instance
         prom_token = os.getenv("PROM_ACCESS_TOKEN")
```
Comment (durandom) [inline metrics/prommetrics.py]: same here, `FLT_` prefix

File: metrics/prommetricslive.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            trigger=IntervalTrigger(minutes=5),
+            id='update_metric_data',
+            name='Ticker to get new data from prometheus',
+            replace_existing=True)
+
+        atexit.register(lambda: scheduler.shutdown()) # Shut down the scheduler when exiting the app
+
+    def _get_metrics_from_prometheus(self, observer=None):
+        # Collect credentials to connect to a prometheus instance
+        prom_token = os.getenv("PROM_ACCESS_TOKEN")
```
Comment (durandom) [inline metrics/prommetricslive.py:42]: and here, `FLT_`

sorry, this slipt my attention before

File: metrics/prommetrics.py [hunk 6]
```diff
@@ -23,43 +29,35 @@ def connect(self):
...
+
+        # Calculate chunk size to download and push to the observer at each instance
+        chunk_seconds = int(round((dateparser.parse('now') - dateparser.parse(chunk_size)).total_seconds()))
+
+        start = dateparser.parse(start_time).timestamp()
+        end = dateparser.parse(end_time).timestamp()
+
+        while start <= end:     # Main loop which iterates through time-ranges to collect a chunk of data at every iteration
+            for metric_name in metrics_list:    # Loop to get a chunk of data for every metric in the list
+                pkt_list = json.loads(prom.get_metric_range_data(metric_name=metric_name, start_time=start, end_time=end))
```
Comment (durandom) [inline metrics/prommetrics.py]: can't we get rid of `json.loads` and `json.dumps` ?
How about `prometheus.py` always returning a `json.loads` dict?

**Review by durandom — CHANGES_REQUESTED** [review]
sorry, I should've been more clear.
I want to trigger the collection of metrics from a prometheus based on ENV variables. If none are set, we'll use the filemetrics from `data/` 
This way you can a) clone the repo and run the `app.py` right away and b) travis will be able to run it.
This approach is sometimes called [feature flags](https://en.wikipedia.org/wiki/Feature_toggle)

File: app.py (hunk 2 above)
Comment (4n4nd) [reply app.py]: Oh that makes a lot of sense. I will look into the feature flags stuff. Thanks

**Comment by durandom** [comment]
you'll have to rebase your branch to current master, because I've changed `app.py`

on your clean branch
```
git remote add upstream https://github.com/AICoE/prometheus-flatliner.git
git fetch upstream
git rebase -i upstream/master 
```

**Pull request closed without merging**